.github/workflows/e2e-manual.yml

```diff
@@ -29,7 +29,7 @@ jobs:
       - name: Generate testnets
         working-directory: test/e2e
         # When changing -g, also change the matrix groups above
-        run: ./build/generator -g 5 -d networks/nightly/
+        run: ./generate-multiversion.sh -g 5 -d networks/nightly/
```

Comment:
So, the `generator` tool in the e2e code generates a suite of test cases to run with different combination of node configurations. Its output is a set of testnet manifests, each of which the `runner` tool then use as a test scenario. Including the multiversion flag here means that the alternate Tendermint version will become another dimension that the generator tool will use when constructing test cases along with the other dimensions such as in vs out-of-process and number of perturbations to run. This means that some of the nodes in some of the testnets will have the alternate version. 

In the case of v0.37 and main, we will _not_ be running an multiversion test because of the logic present in the `generate-multiversion.sh` script. The logic in that script looks for the highest tagged _release_ version on the branch the test is being run from and checks that this version matches `dev` version that the branch is tagged with. In the case of `main` it has tag `v0.38.0-dev` and a most recent tagged release of `v0.34.20`,  (this must have been where we branched to begin development). As a result, `main`'s dev version, `v0.38` does not match `v0.34`, so we will not run multiversion testing on `main`. At the moment, the only time multiversion testing will occur is in the v0.34.x nightly (and manual) e2e tests.

Let me know if this addresses your concern.

Reply:
Yes, I completely understand your intuition here and how it should work. I am just wondering whether we should also not run the same version tests as well. If nothing, it would be useful to have a comment or document it somewhere that by default the testnet will have multiple  (minor)versions of Tendermint.

Reply:
> I am just wondering whether we should also not run the same version tests as well.

I see and I definitely agree. We mostly want to test that the behavior of the new version is correct. This is the rationale behind adding the selection of the alternate version with a lower probability in the testnet generator: https://github.com/tendermint/tendermint/pull/9829/files#diff-40ef656313ea7873bfaf38026d93c96559340552d918492dd3f68c723c04addfR25-R58

For instance, when I generate the testnets on `main` locally, I observe only 10 out of the 48 testnets being generated with nodes of the alternate version.

```
./build/generator --multi-version v0.34.24 --dir ./networks/nightly
[william@sidewinder] tendermint[main]:test/e2e/networks/nightly ◆ rg 'version = "v0.34.24"' -l
gen-0020.toml
gen-0022.toml
gen-0018.toml
gen-0017.toml
gen-0016.toml
gen-0014.toml
gen-0010.toml
gen-0004.toml
gen-0003.toml
gen-0000.toml
```

So we still have many testnets that are only the latest. Because the random seed for generating the e2e tests is fixed, we will always have only these 10 out of 48 being generated and run with multi-versions.